### PR TITLE
FLASH on NERSC

### PR DESCRIPTION
**Pull request by e-carlin**
- Use output of prepare_simulation to start  codes through sbatch with the proper command
- Get FLASH running on NERSC

### CODE REVIEW

File: sirepo/pkcli/job_agent.py [hunk 1]
```diff
@@ -722,17 +737,12 @@ def _processor():
 #SBATCH --output={template_common.RUN_LOG}
 #SBATCH --time={self._sbatch_time()}
 {o}
-cat > bash.stdin <<'EOF'
 {self.job_cmd_source_bashrc()}
 {self.job_cmd_env()}
 if [[ ! $LD_LIBRARY_PATH =~ /usr/lib64/mpich/lib ]]; then
     export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64/mpich/lib
 fi
-#TODO(robnagler) need to get command from prepare_simulation
```
Comment (robnagler) [inline sirepo/pkcli/job_agent.py:731]: yay!

**Review by robnagler — CHANGES_REQUESTED** [review]
Looks good. Only minor comments.

File: sirepo/feature_config.py [hunk 2]
```diff
@@ -114,6 +114,7 @@ def b(msg, dev=False):
         jspec=dict(
             derbenevskrinsky_force_formula=b('Include Derbenev-Skrinsky force formula'),
         ),
+        in_slurm=(False, bool, 'True if being run by slurm'),
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/feature_config.py]: 810919fd5

File: sirepo/mpi.py [hunk 3]
```diff
@@ -55,16 +55,19 @@ def run_program(cmd, output='mpi_run.out', env=None):
         output (str): where to write stdout and stderr
         env (dict): what to pass as env
     """
-    cmd = [
+    m = [
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/mpi.py]: 810919fd5

File: sirepo/pkcli/job_agent.py [hunk 4]
```diff
@@ -582,12 +584,23 @@ def job_cmd_cmd_stdin_env(self, *args, **kwargs):
         return c, s, e
 
     def job_cmd_env(self):
-        e = PKDict()
+        e = PKDict(SIREPO_FEATURE_CONFIG_IN_SLURM=1)
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/pkcli/job_agent.py]: 810919fd5

File: sirepo/package_data/static/json/flash-schema.json [hunk 5]
```diff
@@ -687,6 +687,13 @@
         }
     },
     "model": {
+	"animation": {
+            "jobRunMode": ["Execution Mode", "JobRunMode", "parallel"],
+            "sbatchCores": ["Cores", "Integer", 128],
```
[the remark this answers is not in the document]
Comment (e-carlin) [reply sirepo/package_data/static/json/flash-schema.json:692]: It is. I don't think there is a way to share it currently. Do you have any suggestions?

One thought that has been roaming around my head is wondering if we should build more structure around schemas so we can treat them more as code and less as json. I've been thinking about this wrt the class stuff Mike has been adding and this case seems like another area where we would benefit.

**Comment by e-carlin** [comment]
@robnagler ready for another review.

**Review by robnagler — APPROVED** [review]

**Pull request merged**